sktime/transformations/series/lag.py

```diff
@@ -226,6 +226,8 @@ def _transform(self, X, y=None):
                 Xt.index = X.index + lag
                 X_orig_idx_shifted = X_orig_idx + lag
             else:
+                if hasattr(X.index, "freq") and X.index.freq is None and freq is None:
```

Comment:
hm, did not know that it exists. What would be the benefits?

Reply:
I am surprised that you ask! Normally I get similar suggestions from you ;)

Anyway, `infer_freq` is very general and covers several sktime-related cases. It was created specifically to avoid code repetition and abstract frequency inference.

Reply:
hm, yes, I see.

*but*: `sktime` `infer_freq` seems very useful when we don't know the type of the object (like `ForecastingHorizon` or splitter cases).
In this case, we know for certain that it is a `pandas.DataFrame`, so calling the specific method minimizes coupling.

Reply:
ok, point taken. thanks for the explanation :)

Reply:
well, I didn't really feel strongly about it. Let me know if you think it's better the other way round.